core/src/main/scala/commandcenter/command/ExternalIPCommand.scala

```diff
@@ -11,19 +11,22 @@ final case class ExternalIPCommand(commandNames: List[String]) extends Command[S
   val commandType: CommandType = CommandType.ExternalIPCommand
   val title: String            = "External IP"
 
-  // TODO: Also support Windows (nslookup?). If there's no good solution, making an api.ipify.org request could work too.
-  override val supportedOS: Set[OS] = Set(OS.MacOS, OS.Linux)
-
   def preview(searchInput: SearchInput): ZIO[Env, CommandError, List[PreviewResult[String]]] =
     for {
       input      <- ZIO.fromOption(searchInput.asKeyword).orElseFail(CommandError.NotApplicable)
-      externalIP <- PCommand("dig", "+short", "myip.opendns.com", "@resolver1.opendns.com").string
+      externalIP <- pCommand.string
                       .bimap(CommandError.UnexpectedException, _.trim)
     } yield List(
       Preview(externalIP)
         .score(Scores.high(input.context))
         .onRun(tools.setClipboard(externalIP))
     )
+
+  private def pCommand: PCommand =
+    OS.os match {
+      case OS.Windows => PCommand("powershell", "(Invoke-WebRequest -uri \"api.ipify.org\").Content")
```

Comment:
This works, but I think going down a level to Powershell is better suited for cases that have no other easy alternative.

I'd probably change this method to be `getExternalIP` and use `Sttp` to make the http request for the Windows case while keeping `PCommand` for the others.

I figure this should be faster too. I imagine starting up Powershell has some additional latency.

Reply:
Ok, fair point. Thanks, it's really funky and was an interesting experience to me to learn to work with Powershell as I had never done it before.